projects/igniteui-angular/src/lib/grids/hierarchical-grid/child-grid-row.component.html

```diff
@@ -1,3 +1,3 @@
 <div class="igx-grid__hierarchical-indent" [ngClass]="{'igx-grid__hierarchical-indent--scroll': parentHasScroll}">
-    <igx-hierarchical-grid #hgrid></igx-hierarchical-grid>
+    <igx-hierarchical-grid #hgrid [data]='rowData.childGridsData[layout.key]'></igx-hierarchical-grid>
```

Comment:
When the row is expanded and the parent row reference to object is not updated and only the child data is changed this will not trigger change detection. It changes only when the row is collapsed/expanded again.
I am not sure how this should be detected though.

Reply:
@skrustev Modifying values in the data array will not trigger the hierarchical grid pipes that build the hierarchy, so the change will not be reflected. This is consistent with other features that rely on pipes to process the data. For example, in a normal grid with grouping, changing a value in the array will not in itself re-evaluate the groups. Some other action that triggers the grouping pipe needs to take place for the change to be applied to the groups (expand/collapse, sort etc.).
The behavior is consistent with the normal grid so I don't think we should cover such scenarios.

Reply:
I'm ok with keeping it as it is right now if there are no objections.